fix: Issue #2 - [BUG] Sidebar header text unreadable with certain theme lightness settings

PrayagCodes opened the pull request:
## Description
This PR fixes sidebar text contrast issues in the `Explorer` when users adjust theme lightness settings. Sidebar header and item text colors are now dynamic and adapt to the current theme lightness, ensuring readability across all configurations.

## Before
- video: https://cap.link/11pjpg2ehn4j8x9
<img width="1917" height="943" alt="image" src="https://github.com/user-attachments/assets/8073801c-8bfa-4910-af6a-66b0379a8be9" />

## After
- video: https://cap.link/b7aftqpb07p8hdg
<img width="1918" height="946" alt="image" src="https://github.com/user-attachments/assets/506d8db2-c889-424b-8d26-42c37f116a28" />

## Implementation
- video: https://cap.link/125zrg257vdvhy6
<img width="1901" height="946" alt="image" src="https://github.com/user-attachments/assets/0f7123c3-6c34-4b8b-a73f-6aaae20e21ca" />

## Related Issue
Closes  [Issue #2 - [BUG] Sidebar header text unreadable with certain theme lightness settings ](https://github.com/codepath/puter/issues/6) #6

### Core Functionality
- Made sidebar title text color dynamic (adapts to theme lightness)
- Made sidebar item text color dynamic (adapts to theme lightness)
- Implemented adaptive background colors for active sidebar items
- Made window action button icons (minimize, maximize, close) dynamic
- Fixed CSS variables initialization on app startup

### CSS Variables
- Added `--primary-color-sidebar` variable for future extensibility
- Added `--primary-color-sidebar-item` variable for active item backgrounds
- Added `--window-sidebar-active-bg` variable to link active items to theme
- Added `--window-action-btn-filter` variable for window control buttons

### Theme Service
- Extended `reload_()` method to dynamically set colors based on lightness
- Fixed initialization bug where CSS variables weren't set on first load
- Uses existing `light_text` threshold (< 60) for consistency

## Implementation Details
- Text color switching: `light_text ? 'white' : '#373e44'`
- Active background switching: `light_text ? '#5a5d61aa' : '#fefeff'`
- Icon filter switching: `light_text ? 'invert(1) brightness(2)' : 'none'`
- Uses existing CSS variable infrastructure for consistency
- Threshold at lightness value 60 (consistent with existing theme logic)

## Testing Performed
- [x] Tested at lightness values: 0, 30, 60, 90, 100
- [x] Verified header readability across all lightness values
- [x] Verified sidebar items (Home, Documents, Public, Pictures, Videos) readability
- [x] Tested active item selection contrast on dark themes
- [x] Tested active item selection contrast on light themes
- [x] Verified window control buttons (minimize, maximize, close) visibility
- [x] Tested with both saved theme preferences and default state
- [x] Confirmed no visual regressions on other UI elements

## Files Modified
- `src/gui/src/css/style.css` - CSS variables and styling for sidebar and window controls
- `src/gui/src/services/ThemeService.js` - Dynamic color switching logic and initialization fix

PrayagCodes committed:
fix(ui): improve sidebar text contrast across theme lightness settings-#issue-2

## Problem
When adjusting the lightness level of screen themes, sidebar header texts
and item labels in the explorer become unreadable due to poor contrast
between text and background colors.

## Root Cause
- Sidebar title and item text colors were hardcoded (#8f96a3 and #444444)
- Active item backgrounds were fixed to white (#fefeff)
- No dynamic color adjustment based on theme lightness

## Solution
Implement dynamic contrast adjustment for sidebar text and backgrounds:

### Changes:
- style.css (lines 1225, 1250): Use var(--window-sidebar-color) for text
- style.css (lines 92-93, 106, 1262): Add dynamic active item backgrounds
- ThemeService.js (line 124): Switch colors based on lightness threshold

### Behavior:
- Lightness < 60: White text on dark backgrounds
- Lightness >= 60: Dark text (#373e44) on light backgrounds
- Active items: Contrasting backgrounds in all themes

## Testing
- Verified readability at lightness values: 0, 30, 60, 90, 100
- Tested with 'Favorites' header and all sidebar items
- Confirmed WCAG accessibility compliance

Resolves sidebar readability issues across theme configurations.

PrayagCodes committed:
fix(ui): improve text icon contrast across theme lightness settings

Implement dynamic contrast adjustment for  window controls that
adapts to theme lightness settings.

Changes:
- Window action buttons (minimize, maximize, close) icons adapt to theme
- CSS variables initialized properly on app startup

israx commented on src/gui/src/services/ThemeService.js:
```diff
@@ -87,6 +87,9 @@ export class ThemeService extends Service {
             this.reload_();
+        } else {
+            // Initialize CSS variables with default values
+            this.reload_();
```
Nit: we can remove the else condition and move line 89 below to always run this.reload()

PrayagCodes replied:
resolved

israx reviewed (approved):
Thanks Prayag for the thorough PR description. Looks good to me!

PrayagCodes committed:
refactor: remove redundant else block in ThemeService

Move this.reload_() outside if/else block since it runs in both branches.
This follows DRY principle and simplifies the code logic.

Addresses reviewer nit https://github.com/codepath/puter/pull/21/files/22d494049c588a5da93c0334a7d7b9ae8daf5b3e#r2451999665
